core/trino-main/src/main/java/io/trino/sql/planner/iterative/rule/AddExchangesBelowPartialAggregationOverGroupIdRuleSet.java

```diff
@@ -247,6 +262,25 @@ protected Optional<PlanNode> transform(AggregationNode aggregation, GroupIdNode
                     .map(groupId.getGroupingColumns()::get)
                     .collect(toImmutableList());
 
+            if (isUseHighestCardinalityColumnForForcedExchangeBelowGroupId(context.getSession())) {
+                // use only the symbol with the highest cardinality.
```

Comment:
`Use only the symbol with the highest cardinality (if we have statistics). This makes...`